lib/zendesk_api/resources.rb

```diff
@@ -40,50 +41,7 @@ class Session < ReadResource
     include Destroy
   end
 
-  class Tag < DataResource
-    include Update
-    include Destroy
-
-    alias :name :id
-    alias :to_param :id
-
-    def path(opts = {})
-      raise "tags must have parent resource" unless association.options.parent
-      super(opts.merge(:with_parent => true, :with_id => false))
-    end
-
-    def changed?
-      true
-    end
-
-    def destroy!
-      super do |req|
-        req.body = attributes_for_save
-      end
-    end
-
-    module Update
-      def _save(method = :save)
-        return self unless @resources
-
-        @client.connection.post(path) do |req|
-          req.body = { :tags => @resources.reject(&:destroyed?).map(&:id) }
-        end
-
-        true
-      rescue Faraday::ClientError => e
-        if method == :save
-          false
-        else
-          raise e
-        end
-      end
-    end
-
-    def attributes_for_save
-      { self.class.resource_name => [id] }
-    end
-  end
+  require_relative "resource_tag"
```

Comment:
Of course I don't like `require_relative` in the middle of a page, but sorting out all the circular dependencies started to take too long.